src/terarangerone.cpp

```diff
@@ -124,14 +135,20 @@ void TerarangerOne::serialDataCallback(uint8_t single_character)
         int16_t range = input_buffer[1] << 8;
         range |= input_buffer[2];
 
-        if (range < 14000 && range > 200)
+        if (range < MAX_RANGE_MM && range > MIN_RANGE_MM )
         {
-          range_msg.header.stamp = ros::Time::now();
-          range_msg.header.seq = seq_ctr++;
-          range_msg.range = range * 0.001; // convert to m
-          range_publisher_.publish(range_msg);
+          ros::Time time_now = ros::Time::now();
+          ros::Duration interval = time_now - range_msg_.header.stamp;
+          
+          if ( interval > publish_interval_ )
```

Comment:
check current time against last published message to control publish rate

Reply:
Hello François,

Sorry for the delay, we are currently working on new ROS packages:
-teraranger : package for single sensors
-teraranger_array : package for Tower, Hub and Multiflex

As this package (and others) are gonna be deprecated soon, I will close this PR. However, we greatly appreciate your input and we will try to take it into account as much as we can, which might be tricky because we are changing the serial library.

Best regards,

Pierre-Louis